### DISCUSSION TITLE
Non-Fickian Diffusion

### DISCUSSION THREAD

**Discussion by skydivine22** [PMEAL/OpenPNM#2027] [Q&A] [answered]
Hi, I am wondering whether OpenPNM is able to tackle problems related to non-Fickian diffusion? To my knowledge, all the diffusion simulations are Fickian.

Best

Andy

**Comment by ma-sadeghi** [comment, marked as answer]
Hi Andy,
We don't have "built-in" models for non-Fickian diffusion, but adding one is very straightforward in OpenPNM. You just need to write your own model and replace OpenPNM's default. If you let me know what equation you want to use for calculating the flux (instead of -DAdc/dx), I might be able to help you out write a custom model.
